src/app/authentication/service/signUpService.go

```diff
@@ -1 +1,76 @@
 package service
+
+import (
+	"authentication/constants"
+	"authentication/models"
+	"authentication/utils/db"
+	"log"
+	"net/http"
+	"regexp"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-playground/validator/v10"
+)
+
+var Validate *validator.Validate
+
+func init() {
+	Validate = validator.New()
+}
+
+// validateWithPattern validates a string against a regular expression pattern.
+func validateWithPattern(input string, pattern string) bool {
+	regExp, err := regexp.Compile(pattern)
+	if err != nil {
+		log.Fatalf("%s: %s", constants.ErrCompileRegex, err)
+	}
+	return regExp.MatchString(input)
+}
+
+// ValidatedEmail checks email format.
+func ValidatedEmail(email string) bool {
+	return validateWithPattern(email, constants.ValidatedEmailPattern)
+}
+
+// ValidatePANCard checks PAN card number  format.
+func ValidatePANCard(panCard string) bool {
+	return validateWithPattern(panCard, constants.ValidatePANCardNumber)
+}
+
+// ValidatePassword checks password meets required criteria or not.
+func ValidatePassword(password string) bool {
+	return validateWithPattern(password, constants.ValidatePassword)
+}
+
+// Validation checks for required fields
+func ValidateCustomer(customer models.Customer) error {
+	if err := Validate.Struct(customer); err != nil {
+		return err
+	}
+	return nil
```

Comment:
Remove validation from service and create a validation file inside utils folder and use the validation in handler only.